wgpu-native/src/id.rs

```diff
@@ -0,0 +1,530 @@
+use crate::{Epoch, Id, Index, TypedId};
+#[cfg(feature = "serde")]
+use serde::{Deserialize, Serialize};
+
+#[cfg(not(feature = "gfx-backend-gl"))]
+#[repr(transparent)]
+#[derive(Clone, Copy, Debug, Hash, PartialEq)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct InstanceId(Id);
+
+#[cfg(not(feature = "gfx-backend-gl"))]
+impl InstanceId {
+    fn raw(self) -> Id {
+        self.0
+    }
+}
+
+#[cfg(not(feature = "gfx-backend-gl"))]
+impl TypedId for InstanceId {
```

Comment:
I think we can keep `typed_id` macro since it's not public facing? Unless the other suggestion makes it totally unnecessary.